jenkins/spark-tests.sh

```diff
@@ -125,53 +125,56 @@ export PYTHONPATH="$CONDA_ROOT/lib/python$PYTHON_VER/site-packages:$PYTHONPATH"
 #stop and restart SPARK ETL
-stop-slave.sh
+stop-worker.sh
 stop-master.sh
 start-master.sh
-start-slave.sh spark://$HOSTNAME:7077
+start-worker.sh spark://$HOSTNAME:7077
 jps
 
-export BASE_SPARK_SUBMIT_ARGS="$BASE_SPARK_SUBMIT_ARGS \
---master spark://$HOSTNAME:7077 \
---conf spark.sql.shuffle.partitions=12 \
---conf spark.task.maxFailures=$SPARK_TASK_MAXFAILURES \
---conf spark.dynamicAllocation.enabled=false \
---conf spark.driver.extraJavaOptions=-Duser.timezone=UTC \
---conf spark.executor.extraJavaOptions=-Duser.timezone=UTC \
---conf spark.sql.session.timeZone=UTC"
-
-export SEQ_CONF="--executor-memory 16G \
---total-executor-cores 6"
-
-export PARALLEL_CONF="--executor-memory 4G \
---total-executor-cores 1 \
---conf spark.executor.cores=1 \
---conf spark.task.cpus=1 \
---conf spark.rapids.sql.concurrentGpuTasks=1 \
---conf spark.rapids.memory.gpu.minAllocFraction=0"
-
-export CUDF_UDF_TEST_ARGS="--conf spark.rapids.memory.gpu.allocFraction=0.1 \
---conf spark.rapids.memory.gpu.minAllocFraction=0 \
---conf spark.rapids.python.memory.gpu.allocFraction=0.1 \
---conf spark.rapids.python.concurrentPythonWorkers=2 \
---conf spark.executorEnv.PYTHONPATH=${RAPIDS_PLUGIN_JAR} \
---conf spark.pyspark.python=/opt/conda/bin/python \
---py-files ${RAPIDS_PLUGIN_JAR}"
-
+# BASE spark test configs
+export PYSP_TEST_spark_master=spark://$HOSTNAME:7077
+export PYSP_TEST_spark_sql_shuffle_partitions=12
+export PYSP_TEST_spark_task_maxFailures=$SPARK_TASK_MAXFAILURES
+export PYSP_TEST_spark_dynamicAllocation_enabled=false
+export PYSP_TEST_spark_driver_extraJavaOptions=-Duser.timezone=UTC
+export PYSP_TEST_spark_executor_extraJavaOptions=-Duser.timezone=UTC
+export PYSP_TEST_spark_sql_session_timeZone=UTC
+
+# PARALLEL or non-PARALLEL specific configs
+if [[ $PARALLEL_TEST == "true" ]]; then
+  export PYSP_TEST_spark_cores_max=1
+  export PYSP_TEST_spark_executor_memory=4g
+  export PYSP_TEST_spark_executor_cores=1
+  export PYSP_TEST_spark_task_cores=1
+  export PYSP_TEST_spark_rapids_sql_concurrentGpuTasks=1
+  export PYSP_TEST_spark_rapids_memory_gpu_minAllocFraction=0
+
+  if [[ "${PARALLELISM}" == "" ]]; then
+    PARALLELISM=$(nvidia-smi --query-gpu=memory.free --format=csv,noheader | \
+      awk '{if (MAX < $1){ MAX = $1}} END {print int(MAX / (2 * 1024))}')
+  fi
+  # parallelism > 6 could slow down the whole process, so we have a limitation for it
+  # this is based on our CI gpu types, so we do not put it into the run_pyspark_from_build.sh
+  [[ ${PARALLELISM} -gt 6 ]] && PARALLELISM=6
+  MEMORY_FRACTION=$(python -c "print(1/($PARALLELISM + 0.1))")
+
+  export TEST_PARALLEL=${PARALLELISM}
+  export PYSP_TEST_spark_rapids_memory_gpu_allocFraction=${MEMORY_FRACTION}
+  export PYSP_TEST_spark_rapids_memory_gpu_maxAllocFraction=${MEMORY_FRACTION}
+else
+  export PYSP_TEST_spark_cores_max=6
+  export PYSP_TEST_spark_executor_memory=16g
+  export TEST_PARALLEL=1
```

Comment:
Curious why 1 here?  Everywhere else uses 0 and there's a comment in run_pyspark_from_build stating 0 runs more efficiently.

Reply:
I would also suggest to use TEST_PARALLEL=0 for consistency

Semantically 0 and 1 are equivalent due to this logic

https://github.com/NVIDIA/spark-rapids/blob/d09bc9a7df31872d253af278388b3469ca772a05/integration_tests/run_pyspark_from_build.sh#L169-L173

https://github.com/NVIDIA/spark-rapids/blob/d09bc9a7df31872d253af278388b3469ca772a05/integration_tests/run_pyspark_from_build.sh#L277

Reply:
> Curious why 1 here?

Ah I did not realize its more efficient before, was just trying to make it non-parallel.
Updated to make it consistent